src/robusta/core/sinks/opsgenie/opsgenie_sink.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import re
+
+import opsgenie_sdk
+
+from typing import List
+from tabulate import tabulate
+
+from .opsgenie_sink_params import OpsGenieSinkConfigWrapper
+from ...reporting.blocks import (
+    KubernetesDiffBlock,
+    JsonBlock,
+    MarkdownBlock,
+    HeaderBlock,
+    ListBlock,
+    TableBlock,
+    DividerBlock,
+)
+from ...reporting.base import (
+    Finding,
+    Enrichment,
+    FindingSeverity,
+)
+from ..sink_base import SinkBase
+
+
+PRIORITY_MAP = {
+    FindingSeverity.INFO: "P5",
+    FindingSeverity.LOW: "P4",
+    FindingSeverity.MEDIUM: "P3",
+    FindingSeverity.HIGH: "P1"
+}
+
+
+class OpsGenieSink(SinkBase):
+    def __init__(self, sink_config: OpsGenieSinkConfigWrapper, cluster_name: str):
+        super().__init__(sink_config.opsgenie_sink)
+        self.cluster_name = cluster_name
+        self.api_key = sink_config.opsgenie_sink.api_key
+        self.teams = sink_config.opsgenie_sink.teams
+        self.tags = sink_config.opsgenie_sink.tags
+
+        self.conf = opsgenie_sdk.configuration.Configuration()
+        self.conf.api_key['Authorization'] = self.api_key
+
+        self.api_client = opsgenie_sdk.api_client.ApiClient(configuration=self.conf)
+        self.alert_api = opsgenie_sdk.AlertApi(api_client=self.api_client)
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, OpsGenieSink)
+            and other.sink_name == self.sink_name
+            and other.cluster_name == self.cluster_name
+            and other.api_key == self.api_key
+        )
+
+    def write_finding(self, finding: Finding, platform_enabled: bool):
+        description = self.__to_description(finding, platform_enabled)
+        body = opsgenie_sdk.CreateAlertPayload(
+            source="Robusta",
+            message=finding.title,
+            description=description,
+            alias=finding.title,
+            responders=[
+                {
+                    "name": team,
+                    "type": "team"
+                } for team in self.teams
+            ],
+            tags=self.tags,
+            entity=finding.service_key,
+            priority=PRIORITY_MAP.get(finding.severity, "P3")
+        )
+        try:
+            self.alert_api.create_alert(create_alert_payload=body)
+        except opsgenie_sdk.ApiException as err:
+            logging.error(f"Error creating opsGenie alert {finding} {err}", exc_info=True)
+
+    def __to_description(self, finding: Finding, platform_enabled: bool) -> str:
+        description = ""
+        if platform_enabled:
+            description = f"<a href=\"{finding.investigate_uri}\">Investigate</a>\n"
+
+        return f"{description}{self.__enrichments_as_text(finding.enrichments)}"
+
+    @classmethod
+    def __markdown_to_html(cls, mrkdwn_text: str) -> str:
```

Comment:
You're missing some markdown functionality here like italics, lists, etc.

Does OpsGenie just expect HTML or can it also render some of the markdown stuff?

Can we use a 3rd party library like https://github.com/trentm/python-markdown2 ?

Reply:
OpsGenie expects html only AFAIK.
We're using slack markdown syntax as the base format.

markdown2 converts some of the slack format properly.

I used it and created better conversion